### PR TITLE
USB: netusb: Cleanup and switch to USB transfer API

### LINKED ISSUE #14127: netusb: TX path doesn't work in RNDIS driver
**Describe the bug**
When using the RNDIS driver communication with the host doesn't work. Enabling some debugging shows that the driver is able to receive data, but not to send it. Enabling more debugging shows that the RNDIS driver try to do more USB bulk IN transfer while the current one hasn't finished.

**To Reproduce**
Build any net sample with the RNDIS driver for the interface. Tested on Nucleo L432KC and SAM E70 Xplained.

**Expected behavior**
IP communication works out of the box.

**Impact**
The driver is unusable.

**Screenshots or console output**
```
uart:~$ net ping 192.0.2.2
S[00:00:27.310,000] <dbg> usb_net.netusb_send: Send pkt, len 42
[00:00:27.310,000] <dbg> usb_rndis.rndis_send: send pkt 0x20003968 len 42
[00:00:27.310,000] <dbg> usb_rndis.net_pkt_hexdump: <
[00:00:27.310,000] <dbg> usb_rndis: 
ff ff ff ff ff ff 00 00 |........
5e 00 53 00 08 06       |^.S...  
[00:00:27.310,000] <dbg> usb_rndis: 
00 01 08 00 06 04 00 01 |........
00 00 5e 00 53 00 c0 00 |..^.S...
02 01 00 00 00 00 00 00 |........
c0 00 02 02             |....    
[00:00:27.310,000] <dbg> usb_rndis.rndis_hdr_add: type 1 len 86 payload offset 36 payload len 42
[00:00:27.310,000] <dbg> usb_net: USB <
01 00 00 00 56 00 00 00 |....V...
24 00 00 00 2a 00 00 00 |$...*...
00 00 00 00 00 00 00 00 |........
00 00 00 00 00 00 00 00 |........
00 00 00 00 00 00 00 00 |........
00 00 00 00 ff ff ff ff |........
ff ff 00 00 5e 00 53 00 |....^.S.
08 06 00 01 08 00 06 04 |........
[00:00:27.310,000] <dbg> usb_dc_stm32.usb_dc_ep_write: ep 0x82, len 64
[00:00:27.310,000] <dbg> usb_net: USB <
00 01 00 00 5e 00 53 00 |....^.S.
c0 00 02 01 00 00 00 00 |........
00 00 c0 00 02 02       |......  
[00:00:27.310,000] <dbg> usb_dc_stm32.usb_dc_ep_write: ep 0x82, len 22
[00:00:27.310,000] <err> usb_dc_stm32: Unable to write ep 0x82 (-16)
ent a ping to 192.0.2.2
[00:00:27.310,000] <wrn> usb_net: Error writing to ep 0x82 ret -16
Ping timeout
[00:00:27.310,000] <err> usb_dc_stm32: Unable to write ep 0x82 (-16)
[00:00:27.310,000] <wrn> usb_net: Error writing to ep 0x82 ret -16
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Debian
 - Commit SHA: 35b7fde8a60b55ca0806a50e5b8cb4cca83f1b84

### LINKED ISSUE #14416: [Coverity CID :195789]Uninitialized variables in /subsys/usb/class/netusb/function_rndis.c
Static code scan issues seen in File: /subsys/usb/class/netusb/function_rndis.c
 Category: Uninitialized variables
 Function: append_bytes
 Component: Other
 CID: 195789
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by finikorg**
Cleanup and fixes.

Fixes #14127 
Fixes #14416

### CODE REVIEW

**Review by jfischer-no — APPROVED** [review]

File: subsys/usb/class/netusb/function_rndis.c [hunk 1]
```diff
@@ -222,6 +218,8 @@ static struct __rndis {
 static u8_t manufacturer[] = CONFIG_USB_DEVICE_MANUFACTURER;
 static u32_t drv_version = 1U;
 
+static u8_t tx_buf[1500];
+
```
Comment (aurel32) [inline subsys/usb/class/netusb/function_rndis.c]: This array holds the Ethernet packet, but also the RNDIS header. I guess its size should therefore be `1500 + sizeof(struct rndis_payload_packet)` or even better `NETUSB_MTU + sizeof(struct rndis_payload_packet).

**Review by aurel32 — CHANGES_REQUESTED** [review]
Beside the issue with the TX buffer size, it looks good to me. It's a nice cleanup and I have tested it fixes the issue reported in #14127.

File: subsys/usb/class/netusb/function_rndis.c (hunk 1 above)
Comment (finikorg) [reply subsys/usb/class/netusb/function_rndis.c]: Fixed

**Review by aurel32 — APPROVED** [review]
Thanks for the quick update, LGTM now.

**Review by masz-nordic — APPROVED** [review]

**Comment by finikorg** [comment]
> Beside the issue with the TX buffer size, it looks good to me. It's a nice cleanup and I have tested it fixes the issue reported in #14127.

@aurel32 Have you tested it on Windows? For me it works in Windows but not in Linux, I suspect this is issue in stm32 USB controller driver.

**Comment by aurel32** [comment]
@finikorg Nope I only tested it on Linux (4.19 kernel), with a SAM E70 Xplained board. I can try with an STM32 board in the next days.

**Pull request merged**